src/main/java/duke/storage/Storage.java

```diff
@@ -0,0 +1,127 @@
+    private final String SEPARATE = " / ";
+
+    /**
+     * Constructor to initialise the Storage.
+     * @param pathName The path name to be converted to create a new File instance.
+     */
+    public Storage(String pathName) {
+        this.file = new File(pathName);
+        if (!this.file.getParentFile().exists()) {
+            this.file.getParentFile().mkdirs();
+        }
+    }
+
+    /**
+     * Saves the contents of the tasklist to the designated file.
+     * @param taskList The current state of the tasklist handled by Duke.
+     */
+    public void saveFile(ArrayList<Task> taskList) throws IOException {
+        try {
+            FileWriter myWriter = new FileWriter(this.file);
+            String taskToFile = "";
+
+            // Iterate through every element in the tasklist to create the text version of the task
+            for (Task task : taskList) {
+                String checkTaskType = task.toString();
+                String taskStatus = (task.getStatusIcon().equals("X")) ? "1" : "0";
+
+                // Determine the type of task
+                switch(checkTaskType.charAt(1)) {
+                case 'T':
+                    Todo todo = (Todo) task;
+                    taskToFile = "Todo" + SEPARATE + taskStatus + SEPARATE + todo.getTaskDescription();
+                    break;
+                case 'D':
+                    Deadline deadline = (Deadline) task;
+                    taskToFile = "Deadline" + SEPARATE + taskStatus + SEPARATE + deadline.getTaskDescription()
+                            + SEPARATE + deadline.getBy();
+                    break;
+                case 'E':
+                    Event event = (Event) task;
+                    taskToFile = "Event" + SEPARATE + taskStatus + SEPARATE + event.getTaskDescription()
+                            + SEPARATE + event.getAt();
+                }
+
+                // Save the newly text version of the task into the file
+                myWriter.write(taskToFile + "\n");
+            }
+
+            myWriter.close();
+
+        } catch (IOException e) {
+            System.out.println("Master something went wrong, I can't save the Tasklist! ☹");
+            e.printStackTrace();
+        }
+    }
+
+    /**
+     * Loads the contents of the previously saved tasklist from the designated file.
+     * @return The current state of the tasklist handled by Duke.
+     * @throws IOException
+     */
+    public ArrayList<Task> loadFile() throws IOException {
+        try {
+            Scanner scanner = new Scanner(file);
+            ArrayList<Task> taskList = new ArrayList<>();
+
+            // Iterate through line by line of the designated file to obtain every task in text form
+            while (scanner.hasNextLine()) {
+                String currTask = scanner.nextLine();
+                Task task = null;
+                String[] taskBreakdown = currTask.split(SEPARATE);
```

Comment:
Maybe the naming of the array here should be something to suggest the data structure used is an array?